grails-datastore-gorm/src/main/groovy/org/grails/datastore/gorm/bootstrap/AbstractDatastoreInitializer.groovy

```diff
@@ -296,6 +298,30 @@ abstract class AbstractDatastoreInitializer implements ResourceLoaderAware{
                     datastore = ref("${type}Datastore")
                 }
             }
+            final SoftServiceLoader<Service> services = SoftServiceLoader.load(Service)
+            for (ServiceDefinition<Service> serviceDefinition: services) {
+                if (serviceDefinition.isPresent()) {
+                    final Class<Service> clazz = serviceDefinition.getType()
+                    if (clazz.simpleName.startsWith('$')) {
+                        String serviceClassName = clazz.name - '$' - 'Implementation'
+                        final ClassLoader cl = org.grails.datastore.mapping.reflect.ClassUtils.classLoader
+                        final Class<?> serviceClass = cl.loadClass(serviceClassName)
+
+                        final grails.gorm.services.Service ann = clazz.getAnnotation(grails.gorm.services.Service)
+                        String serviceName = ann?.name()
+                        if(serviceName == null) {
+                            serviceName = Introspector.decapitalize(serviceClass.simpleName)
+                        }
+                        if (serviceClass != null && serviceClass != Object.class) {
+                            "$serviceName"(GormMethodInvokingFactoryBean) {
```

Comment:
You need to ensure the service is autowired by name

Reply:
Did you mean to test if it works, or is there some specific code I need to add which I am not aware of? 

I added the test in the example test-data-service project for each underline GORM implementation. Please check [TestSerivceSpec](https://github.com/grails/gorm-hibernate5/blob/11207/examples/test-data-service/src/integration-test/groovy/example/TestServiceSpec.groovy#L32).

However, I am not sure if the bean is autowired by name. From looking at ServiceRegistryFactoryBean, I noticed that we calling `autowireCapableBeanFactory?.autowireBean(service)` but I do not know how to achieve the same in this case.